### PR TITLE
feature: ignored upstream nodes with a weight of 0.

### PR DESCRIPTION
**Pull request by membphis**

### CODE REVIEW

**Comment by moonming** [comment]
fixed https://github.com/iresty/apisix/issues/532

File: lua/apisix/balancer.lua [hunk 1]
```diff
@@ -47,18 +47,27 @@ end
...
 local function fetch_health_nodes(upstream, checker)
+    local up_nodes = core.table.new(0, #upstream.nodes)
+
     if not checker then
+        for addr, weight in pairs(upstream.nodes) do
+            if weight > 0 then
+                up_nodes[addr] = weight
+            end
+        end
         return upstream.nodes
```
Comment (chnliyong) [inline lua/apisix/balancer.lua]: Here should `return up_nodes`

File: lua/apisix/balancer.lua [hunk 2]
```diff
@@ -47,18 +47,27 @@ end
...
     end
 
     local host = upstream.checks and upstream.checks.host
     local up_nodes = core.table.new(0, #upstream.nodes)
```
Comment (chnliyong) [inline lua/apisix/balancer.lua]: Duplicate line?

File: t/node/chash-balance.t [hunk 3]
```diff
@@ -171,3 +171,81 @@ GET /t
...
+                 [[{
+                    "uri": "/server_port",
+                    "upstream": {
+                        "key": "remote_addr",
+                        "type": "chash",
+                        "nodes": {
+                            "127.0.0.1:1980": 1,
+                            "127.0.0.1:1981": 0,
+                            "127.0.0.1:1982": 0
+                        }
```
Comment (moonming) [inline t/node/chash-balance.t]: need to check the `nodes` wrote into etcd.

File: t/node/rr-balance.t [hunk 4]
```diff
@@ -246,3 +246,80 @@ GET /t
...
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                    "uri": "/server_port",
+                    "upstream": {
+                        "type": "roundrobin",
+                        "nodes": {
+                            "127.0.0.1:1980": 3,
+                            "127.0.0.1:1981": 0,
```
Comment (moonming) [inline t/node/rr-balance.t]: the admin API will return `127.0.0.1:1981` or not? we need add test cases for that.

File: lua/apisix/balancer.lua (hunk 1 above)
Comment (membphis) [reply lua/apisix/balancer.lua]: old implementation is enough, it support weight equal 0 case already. we do not need to patch here, I reverted those new code.

File: lua/apisix/balancer.lua (hunk 2 above)
Comment (membphis) [reply lua/apisix/balancer.lua]: ditto

File: t/node/rr-balance.t (hunk 4 above)
Comment (membphis) [reply t/node/rr-balance.t]: https://github.com/iresty/apisix/pull/536/files#diff-3d55f969c5c7e4abd188c26251844464R750

added new test case.

File: t/node/chash-balance.t (hunk 3 above)
Comment (membphis) [reply t/node/chash-balance.t]: https://github.com/iresty/apisix/pull/536/files#diff-3d55f969c5c7e4abd188c26251844464R750

fixed

**Pull request merged**